jpegexport.py

```diff
@@ -122,76 +125,75 @@ def processSelected(self):
         
 
         if self.options.fast == True:  # uses simpletransform
-            bbox = simpletransform.computeBBox(nodelist)
-            startx = math.ceil(bbox[0])
-            endx = math.ceil(bbox[1])
-            h = -bbox[2] + bbox[3]
-            starty = toty - math.ceil(bbox[2]) -h
-            endy = toty - math.ceil(bbox[2])
+            bbox = sum([node.bounding_box() for node in nodelist], None)
+            #inkex.utils.debug(bbox) - see transform.py
+            '''
+             width = property(lambda self: self.x.size)
+             height = property(lambda self: self.y.size)
+             top = property(lambda self: self.y.minimum)
+             left = property(lambda self: self.x.minimum)
+             bottom = property(lambda self: self.y.maximum)
+             right = property(lambda self: self.x.maximum)
+             center_x = property(lambda self: self.x.center)
+             center_y = property(lambda self: self.y.center)
+            '''
+            startx = math.ceil(bbox.left)
+            endx = math.ceil(bbox.right)
+            h = -bbox.top + bbox.bottom
+            starty = toty - math.ceil(bbox.top) -h
+            endy = toty - math.ceil(bbox.top)
 
         coords = [startx / scale, starty / scale, endx / scale, endy / scale]
         return coords
 
     def exportArea(self, x0, y0, x1, y1, curfile, outfile, bgcol):
         tmp = self.getTmpPath()
-        command="inkscape -a %s:%s:%s:%s -d %s -e \"%sjpinkexp.png\" -b \"%s\" %s" % (x0, y0, x1, y1, self.options.density, tmp, bgcol, curfile)
-
-        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        command="inkscape --export-area %s:%s:%s:%s -d %s --export-filename \"%sjpinkexp.png\" -b \"%s\" \"%s\"" % (x0, y0, x1, y1, self.options.density, tmp, bgcol, curfile)
```

Comment:
Inkscape also has export functions that will use the same Inkscape and environment that is currently in use. I couldn't make it work on Windows, but here's a link that uses it and works on Linux:

https://gitlab.com/Moini/nextgenerator/-/blob/master/next_gen.py#L125

Reply:
(If you figure out why this doesn't work on Windows, I'd be very glad for the help, see https://gitlab.com/Moini/nextgenerator/-/issues/5 )